modelforge/dataset/spice114.py

```diff
@@ -0,0 +1,280 @@
+            If set to True, a subset of the dataset is used for unit testing purposes; by default False.
+        local_cache_dir: str, optional
+            Path to the local cache directory, by default ".".
+        force_download: bool, optional
+            If set to True, we will download the dataset even if it already exists; by default False.
+        regenerate_cache: bool, optional
+            If set to True, we will regenerate the npz cache file even if it already exists, using
+            the data from the hdf5 file; by default False.
+        Examples
+        --------
+        >>> data = SPICE2Dataset()  # Default dataset
+        >>> test_data = SPICE2Dataset(for_unit_testing=True)  # Testing subset
+        """
+
+        _default_properties_of_interest = [
+            "geometry",
+            "atomic_numbers",
+            "dft_total_energy",
+            "dft_total_force",
+            "mbis_charges",
+        ]  # NOTE: Default values
+
+        self._properties_of_interest = _default_properties_of_interest
+        if for_unit_testing:
+            dataset_name = f"{dataset_name}_subset"
+
+        self.dataset_name = dataset_name
+        self.for_unit_testing = for_unit_testing
+
+        from openff.units import unit
+
+        # SPICE provides reference values that depend upon charge, as charged molecules are included in the dataset.
+        # The reference_energy (i.e., sum of the value of isolated atoms with appropriate charge considerations)
+        # are included in the dataset, along with the formation_energy, which is the difference between
+        # the dft_total_energy and the reference_energy.
+
+        # To be able to use the dataset for training in a consistent way with the ANI datasets, we will only consider
+        # the ase values for the uncharged isolated atoms, if available. Ions will use the values Ca 2+, K 1+, Li 1+, Mg 2+, Na 1+.
+        # See spice_2_curation.py for more details.
+
+        # We will need to address this further later to see how we best want to handle this; the ASE are just meant to bring everything
+        # roughly to the same scale, and values do not vary substantially by charge state.
+
+        # Reference energies, in hartrees, computed with Psi4 1.5 wB97M-D3BJ/def2-TZVPPD.
+
+        self._ase = {
+            "Br": -2574.1167240829964 * unit.hartree,
+            "C": -37.87264507233593 * unit.hartree,
+            "Ca": -676.9528465198214 * unit.hartree,  # 2+
+            "Cl": -460.1988762285739 * unit.hartree,
+            "F": -99.78611622985483 * unit.hartree,
+            "H": -0.498760510048753 * unit.hartree,
+            "I": -297.76228914445625 * unit.hartree,
+            "K": -599.8025677513111 * unit.hartree,  # 1+
+            "Li": -7.285254714046546 * unit.hartree,  # 1+
+            "Mg": -199.2688420040449 * unit.hartree,  # 2+
+            "N": -54.62327513368922 * unit.hartree,
+            "Na": -162.11366478783253 * unit.hartree,  # 1+
```

Comment:
If we take carbon for example, it could be between about 0.05 and 0.4 hartrees depending on the charged state, or about 30 to 110 kj/mol, roughly speaking.  

`            "C": {-1: -37.91424135791358, 0: -37.87264507233593, 1: -37.45349214963933},
`
That is not necessarily a small difference, but pretty small considering the scale (less than 2% difference in the values). If we did the linear regression on any of the datasets with different subsets of the data, we'd probably have larger variability (as we've talked a bunch about in issue #72).  Taking into this into account would mean we'd have to be passing the charge info when we remove self energies and when we do inference to  add the "right" values back in.  Given the overall goal of removing the self-energies, I don't think this is something we need to worry about.